### PR TITLE
Updated CONTRIBUTING.MD

### LINKED ISSUE #6965: Update Contributing Guidelines and Readme to explain lockbot
Summary:
- Issues closed over 90 days with no more activity will lock automatically
- In an effort to keep issues up to date, we welcome anybody to open new issues if a past locked issue was only partially solved. 
    - New issues addressing locked issues should reference them. 
- A locked issue does not mean an unimportant issue, we are attempting to help keep an up-to-date record of what is still a relevant issue.

@jasongrout I'm working on updating the markdown files, did I miss anything from this morning's discussion?

### PR DESCRIPTION
**Pull request by tgeorgeux**
## References

Fixes #6965

## Code changes

None

## User-facing changes

Updates CONTRIBUTING.md and README.md

## Backwards-incompatible changes

None

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -44,10 +44,17 @@ a keyboard shortcut or automatically on save.
...
 Generally, an issue should be opened describing a piece of proposed work and the
-issues it solves before a pull request is opened. This lets community members
-participate in the design discussion, makes others aware of work being done, and
-sets the stage for a fruitful community interaction. A pull request should
-reference the issue it is addressing.
+issues it solves before a pull request is opened. 
+
+### Issue Management
+Opening an issue lets community members participate in the design discussion, 
+makes others aware of work being done, andsets the stage for a fruitful community
```
Comment (jasongrout) [inline CONTRIBUTING.md]:
```suggestion
makes others aware of work being done, and sets the stage for a fruitful community 
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -44,10 +44,17 @@ a keyboard shortcut or automatically on save.
...
-sets the stage for a fruitful community interaction. A pull request should
-reference the issue it is addressing.
+issues it solves before a pull request is opened. 
+
+### Issue Management
+Opening an issue lets community members participate in the design discussion, 
+makes others aware of work being done, andsets the stage for a fruitful community 
+interaction. A pull request should reference the issue it is addressing. Once the 
+pull request is merged, the issue related to it will also be closed. If there is 
+additional discussion around implemementation the issue may be re-opened. Once 90 days
```
Comment (jasongrout) [inline CONTRIBUTING.md]:
```suggestion
additional discussion around implemementation the issue may be re-opened. Once 30 days
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -44,10 +44,17 @@ a keyboard shortcut or automatically on save.
...
+have passed with no additional discussion '@lockbot' will permanently lock the issue.
```
Comment (jasongrout) [inline CONTRIBUTING.md]:
```suggestion
have passed with no additional discussion '@lockbot' will lock the issue.
```

(we can always unlock the issue)

Comment (tgeorgeux) [reply CONTRIBUTING.md]: Ah, I didn't know that :)

File: README.md [hunk 4]
```diff
@@ -152,7 +152,7 @@ To be listed, please submit a pull request with your information.
 
 We encourage you to ask questions on the [Discourse forum](https://discourse.jupyter.org/c/jupyterlab). A question answered there can become a useful resource for others.
 
-Please use the [GitHub issues page](https://github.com/jupyterlab/jupyterlab/issues) to provide feedback or submit a bug report.
+Please use the [GitHub issues page](https://github.com/jupyterlab/jupyterlab/issues) to provide feedback or submit a bug report. In an effort to keep up with lots of feedback, '@lock[bot]' may lock an issue that isn't fully resolved. If this happened to an issue important to you please open a new issue referencing the locked issue to update the community.
```
Comment (jasongrout) [inline README.md]:
```suggestion
Please use the [GitHub issues page](https://github.com/jupyterlab/jupyterlab/issues) to provide feedback or submit a bug report. To keep resolved issues self-contained, a bot will lock closed issues as resolved after a period of inactivity. If related discussion is still needed after an issue is locked, please open a new issue and reference the old issue.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -44,10 +44,17 @@ a keyboard shortcut or automatically on save.
...
-reference the issue it is addressing.
+issues it solves before a pull request is opened. 
+
+### Issue Management
+Opening an issue lets community members participate in the design discussion, 
+makes others aware of work being done, and sets the stage for a fruitful community 
+interaction. A pull request should reference the issue it is addressing. Once the 
+pull request is merged, the issue related to it will also be closed. If there is 
+additional discussion around implemementation the issue may be re-opened. Once 30 days
+have passed with no additional discussion '@lock[bot]' will lock the issue. If
```
Comment (jasongrout) [inline CONTRIBUTING.md]:
```suggestion
have passed with no additional discussion, the [lock bot](https://github.com/apps/lock) will lock the issue. If
```

**Comment by jasongrout** [comment]
I suggested a few tweaks to the wording.

File: README.md (hunk 4 above)
Comment (tgeorgeux) [reply README.md]: Looks good to me.

**Comment by tgeorgeux** [comment]
Those look good. Thank you.

**Pull request merged**